src/intro_movie_scene_1.c

```diff
@@ -0,0 +1,431 @@
+        if (timer >= 1) {
+            IntroMovie_StartBlendFadeEffect(&bgAnimCnt->blend[0], 1, 0x1E, 0x50, 1, 1);
+            IntroMovie_StartBlendFadeEffect(&bgAnimCnt->blend[1], 2, 0x1E, 0x50, 1, 0);
+            IntroMovie_StartBgScroll_VBlank(bgConfig, bgAnimCnt->scroll, GF_BG_LYR_MAIN_1, 0, -0x20, 0xF0);
+            IntroMovie_StartBgScroll_VBlank(bgConfig, bgAnimCnt->scroll, GF_BG_LYR_MAIN_2, 0, -0x10, 0xF0);
+            IntroMovie_StartSpriteAnimAndMakeVisible(sceneData->sunSprite, TRUE);
+            IntroMovie_AdvanceSceneStep(data);
+        }
+        break;
+    case 7:  // Wait blend and scroll effects (240 frames)
+        if (bgAnimCnt->blend[0].finished && bgAnimCnt->blend[1].finished && IntroMovie_WaitBgScrollAnim(bgAnimCnt->scroll, GF_BG_LYR_MAIN_1) && IntroMovie_WaitBgScrollAnim(bgAnimCnt->scroll, GF_BG_LYR_MAIN_2)) {
+            IntroMovie_AdvanceSceneStep(data);
+        }
+        break;
+    case 8:  // Delay 128 frames then start box legendary anim
+        if (timer >= 128) {
+            IntroMovie_StartSpriteAnimAndMakeVisible(sceneData->birdSprite, TRUE);
+            IntroMovie_AdvanceSceneStep(data);
+        }
+        break;
+    case 9:  // Run anim 90 frames then start fade to white
+        if (timer >= 90) {
+            BeginNormalPaletteFade(0, 0, 0, RGB_WHITE, 65, 1, HEAP_ID_INTRO_MOVIE);
+            IntroMovie_AdvanceSceneStep(data);
+        }
+        break;
+    case 10:  // Wait fade to white (65 frames)
+        if (IsPaletteFadeFinished()) {
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+void IntroMovie_Scene1_Exit(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData) {
+    BgConfig *bgConfig = IntroMovie_GetBgConfig(data);
+    Main_SetVBlankIntrCB(NULL, NULL);
+    if (sceneData->needFreeGfx) {
+        IntroMovie_Scene1_DestroySpritesAndObjectGfx(data, sceneData);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_SUB_0);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_SUB_1);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_SUB_2);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_SUB_3);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_MAIN_0);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_MAIN_1);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_MAIN_2);
+        FreeBgTilemapBuffer(bgConfig, GF_BG_LYR_MAIN_3);
+        sceneData->needFreeGfx = FALSE;
+    }
+}
+
+void IntroMovie_Scene1_InitBgs(IntroMovieOvyData *data) {
+    BgConfig *bgConfig = IntroMovie_GetBgConfig(data);
+
+    {
+        GraphicsModes graphicsModes = {
+            GX_DISPMODE_GRAPHICS,
+            GX_BGMODE_0,
+            GX_BGMODE_0,
+            GX_BG0_AS_2D
+        };
+        SetBothScreensModesAndDisable(&graphicsModes);
+    }
+
+    {
+        BgTemplate bgTemplate =  {
+            0, 0,
+            GF_BG_BUF_SIZE_256x256_4BPP,
+            0x00000000,
+            GF_BG_SCR_SIZE_256x256,
+            GX_BG_COLORMODE_16,
+            0,
+            1,
+            0,
```

Comment:
const, also our standard for BgTemplates is to ``.var = val`` each element of the struct

Reply:
Just for BG templates? Or for all structs?

Reply:
Preferably for all structs, but it's already established for BgTemplates/I only care about BgTemplates for this pr.